### PR TITLE
fun with erased_serde

### PR DESCRIPTION
**Pull request by obmarg**

### CODE REVIEW

File: crux_core/src/bridge/mod.rs [hunk 1]
```diff
@@ -44,35 +42,66 @@ where
...
+        let mut return_buffer = vec![];
+        let mut deser = bincode::Deserializer::from_slice(
+            event,
+            DefaultOptions::new()
+                .with_fixint_encoding()
+                .allow_trailing_bytes(),
+        );
+        self.inner.process_event(
+            &mut deser,
+            &mut bincode::Serializer::new(&mut return_buffer, DefaultOptions::new()),
```
Comment (obmarg) [inline crux_core/src/bridge/mod.rs:66]: This probably shouldn't be default options actually, but the `with_fixint_encoding` I copied from above started complaining and I couldn't be bothered figuring it out

Comment (charypar) [reply crux_core/src/bridge/mod.rs:66]: Should they both be default options instead...? Do we have opinions on the bincode serialization?

**Review by charypar — APPROVED** [review]
This is great. Only thing I'd change is a few names here and there, but looks exactly like what I was trying, but the compiler wouldn't let me have it without`erased_serde`.

Good Stuff! 👏🏻

File: crux_core/src/bridge/mod.rs (hunk 1 above)
Comment (obmarg) [reply crux_core/src/bridge/mod.rs:66]: These are the options that `bincode::deserialize` sets up, so I'm just keeping things consistent.  Bit weird of that they don't use default options by default, but... y'know

Comment (obmarg) [reply crux_core/src/bridge/mod.rs:66]: Although looks like I didn't push up the change where I fixed this. hmm.

Comment (obmarg) [reply crux_core/src/bridge/mod.rs]: Ok, options are now consistently the same as those used by `bincode::deserialize`

**Pull request merged**